palette/src/oklab.rs

```diff
@@ -0,0 +1,792 @@
+    pub b: T,
+
+    /// The white point associated with the color's illuminant and observer.
+    /// D65 for 2 degree observer is used by default.
+    #[cfg_attr(feature = "serializing", serde(skip))]
+    #[palette(unsafe_zero_sized)]
+    pub white_point: PhantomData<Wp>,
+}
+
+impl<Wp, T> Copy for Oklab<Wp, T>
+where
+    T: FloatComponent,
+    Wp: WhitePoint,
+{
+}
+
+impl<Wp, T> Clone for Oklab<Wp, T>
+where
+    T: FloatComponent,
+    Wp: WhitePoint,
+{
+    fn clone(&self) -> Oklab<Wp, T> {
+        *self
+    }
+}
+
+impl<Wp, T> Oklab<Wp, T>
+where
+    Wp: WhitePoint,
+    T: FloatComponent,
+{
+    /// Create an Oklab color.
+    pub fn new(l: T, a: T, b: T) -> Self {
+        Self {
+            l,
+            a,
+            b,
+            white_point: PhantomData,
+        }
+    }
+
+    /// Convert to a `(L, a, b)` tuple.
+    pub fn into_components(self) -> (T, T, T) {
+        (self.l, self.a, self.b)
+    }
+
+    /// Convert from a `(L, a, b)` tuple.
+    pub fn from_components((l, a, b): (T, T, T)) -> Self {
+        Self::new(l, a, b)
+    }
+
+    /// Return the `l` value minimum.
+    pub fn min_l() -> T {
+        from_f64(0.0)
+    }
+
+    /// Return the `l` value maximum.
+    pub fn max_l() -> T {
+        from_f64(1.0)
+    }
+
+    /// Return the `a` value minimum.
+    pub fn min_a() -> T {
+        from_f64(-1.0)
+    }
+
+    /// Return the `a` value maximum.
+    pub fn max_a() -> T {
+        from_f64(1.0)
+    }
+
+    /// Return the `b` value minimum.
+    pub fn min_b() -> T {
+        from_f64(-1.0)
+    }
+
+    /// Return the `b` value maximum.
+    pub fn max_b() -> T {
+        from_f64(1.0)
+    }
+}
+
+///<span id="Oklaba"></span>[`Oklaba`](crate::Oklaba) implementations.
+impl<Wp, T, A> Alpha<Oklab<Wp, T>, A>
+where
+    Wp: WhitePoint,
+    T: FloatComponent,
+    A: Component,
+{
+    /// Oklab and transparency.
+    pub fn new(l: T, a: T, b: T, alpha: A) -> Self {
+        Alpha {
+            color: Oklab::new(l, a, b),
+            alpha,
+        }
+    }
+
+    /// Convert to a `(L, a, b, alpha)` tuple.
+    pub fn into_components(self) -> (T, T, T, A) {
+        (self.l, self.a, self.b, self.alpha)
+    }
+
+    /// Convert from a `(L, a, b, alpha)` tuple.
+    pub fn from_components((l, a, b, alpha): (T, T, T, A)) -> Self {
+        Self::new(l, a, b, alpha)
+    }
+}
+
+impl<Wp, T> FromColorUnclamped<Oklab<Wp, T>> for Oklab<Wp, T>
+where
+    Wp: WhitePoint,
+    T: FloatComponent,
+{
+    fn from_color_unclamped(color: Self) -> Self {
+        color
+    }
+}
+
+/// This implementation is **only valid if the XYZ white point is D65**.
+impl<Wp, T> FromColorUnclamped<Xyz<Wp, T>> for Oklab<Wp, T>
+where
+    Wp: WhitePoint,
+    T: FloatComponent,
+{
+    fn from_color_unclamped(color: Xyz<Wp, T>) -> Self {
+        let l = from_f64::<T>(M1[0][0]) * color.x
+            + from_f64::<T>(M1[0][1]) * color.y
+            + from_f64::<T>(M1[0][2]) * color.z;
```

Comment:
There's a module with matrix operations in `src/matrix.rs`.

Same comment for the changes in `xyz.rs`.